pkg/virt-controller/watch/migration.go

```diff
@@ -1968,3 +1973,28 @@ func (c *MigrationController) removeHandOffKey(migrationKey string) {
 
 	delete(c.handOffMap, migrationKey)
 }
+
+func (c *MigrationController) handleCPUHotplugTargetPod(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) {
+	if vmi.IsCPUDedicated() {
+		var cpuLimit resource.Quantity
+		for _, container := range pod.Spec.Containers {
+			if container.Name == "compute" {
+				if limit, ok := container.Resources.Limits[k8sv1.ResourceCPU]; ok {
+					cpuLimit = limit
+				}
+			}
+		}
+		setCPULimitAnnotationOnVMI(cpuLimit, vmi)
+
+	}
+}
```

Comment:
Further to our [previous conversation](https://github.com/kubevirt/kubevirt/pull/9757#discussion_r1216642647), I think workload-updater should be doing that